projects/miopen/driver/adam_driver.hpp

```diff
@@ -571,15 +684,35 @@ int AdamDriver<Tgpu, Tref, Tgrad>::VerifyForward()
 {
     RunForwardCPU();
     const Tref tolerance = GetTolerance();
-    auto error           = miopen::rms_range(param_host, param);
 
-    if(!std::isfinite(error) || error > tolerance)
+    if(use_multithread)
     {
-        std::cout << "Forward Adam FAILED: " << error << std::endl;
-        return EC_VerifyFwd;
+        const auto error = miopen::rms_range(param_host, param);
+
+        if(!std::isfinite(error) || error > tolerance)
+        {
+            std::cout << "Forward Adam FAILED on updated multi-threaded CPU reference: " << error
+                      << std::endl;
+            return EC_VerifyFwd;
+        }
+
+        std::cout << "Forward Adam Verifies OK on updated multi-threaded CPU reference"
+                  << std::endl;
     }
+    else
+    {
+        const auto error = miopen::rms_range(param_host, param);
 
-    std::cout << "Forward Adam Verifies OK on CPU reference" << std::endl;
+        if(!std::isfinite(error) || error > tolerance)
+        {
+            std::cout << "Forward Adam FAILED on updated single-threaded CPU reference: " << error
+                      << std::endl;
+            return EC_VerifyFwd;
+        }
+
+        std::cout << "Forward Adam Verifies OK on updated single-threaded CPU reference"
+                  << std::endl;
+    }
```

Comment:
There seems to be duplication here as well, consider something like this for logging.

```suggestion
    const auto error = miopen::rms_range(param_host, param);
    const char* ref_type =
        use_multithread ? "multi-threaded" : "single-threaded";

    if(!std::isfinite(error) || error > tolerance)
    {
        std::cout << "Forward Adam FAILED on updated " << ref_type
                  << " CPU reference: " << error << std::endl;
        return EC_VerifyFwd;
    }

    std::cout << "Forward Adam Verifies OK on updated " << ref_type
              << " CPU reference" << std::endl;
```